src/algebra/module/localized_module.lean

```diff
@@ -387,4 +387,280 @@ instance localized_module_is_localized_module :
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (m : M) (s : S) :
+  localized_module.lift S g h (localized_module.mk m s) = (h s).unit⁻¹.1 (g m) := rfl
+
+lemma _root_.localized_module.lift_comp (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
+  (localized_module.lift S g h).comp (localized_module.mk_linear_map S M) = g :=
+begin
+  ext x, dsimp, rw localized_module.lift_mk,
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, one_smul],
+end
+
+lemma _root_.localized_module.lift_unique (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (l : localized_module S M →ₗ[R] M'')
+  (hl : l.comp (localized_module.mk_linear_map S M) = g) :
+  localized_module.lift S g h = l :=
+begin
+  ext x, induction x using localized_module.induction_on with m s,
+  rw [localized_module.lift_mk],
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, ←hl, linear_map.coe_comp, function.comp_app,
+    localized_module.mk_linear_map_apply, ←l.map_smul, localized_module.smul'_mk],
+  congr' 1, rw localized_module.mk_eq,
+  refine ⟨1, _⟩, simp only [one_smul], refl,
+end
+
+/--
+If `M'` is a localized module and `g` is a linear map `M' → M''` such that all scalar multiplication
+by `s : S` is invertible, then there is a linear map `M' → M''`.
+-/
+noncomputable def lift (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
+  M' →ₗ[R] M'' :=
+(localized_module.lift S g h).comp (iso S f).symm.to_linear_map
+
+lemma lift_comp (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
+  (lift S f g h).comp f = g :=
+begin
+  dunfold is_localized_module.lift,
+  rw [linear_map.comp_assoc],
+  convert localized_module.lift_comp S g h,
+  exact iso_symm_comp _ _,
+end
+
+lemma lift_unique (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (l : M' →ₗ[R] M'') (hl : l.comp f = g) :
+  lift S f g h = l :=
+begin
+  dunfold is_localized_module.lift,
+  rw [localized_module.lift_unique S g h (l.comp (iso S f).to_linear_map), linear_map.comp_assoc,
+    show (iso S f).to_linear_map.comp (iso S f).symm.to_linear_map = linear_map.id, from _,
+    linear_map.comp_id],
+  { rw [linear_equiv.comp_to_linear_map_symm_eq, linear_map.id_comp], },
+  { rw [linear_map.comp_assoc, ←hl], congr' 1, ext x,
+    erw [from_localized_module_mk, module.End_algebra_map_is_unit_inv_apply_eq_iff, one_smul], },
+end
+
+/--
+Universal property from localized module:
+If `(M', f : M ⟶ M')` is a localized module then it satisfies the following universal property:
+For every `R`-module `M''` which every `s : S`-scalar multiplication is invertible and for every
+`R`-linear map `g : M ⟶ M''`, there is a unique `R`-linear map `l : M' ⟶ M''` such that
+`l ∘ f = g`.
+```
+M -----f----> M'
+|           /
+|g       /
+|     /   l
+v   /
+M''
+```
+-/
+lemma is_universal :
```

Comment:
Can you check if this is the usual name used in the library?

Reply:
In ring_theory folder, it is called `ring_hom_ext` which says a slightly different thing, I will keep this `is_universal` lemma and add a `ring_hom_ext`.